lib/terraform/runner/request.rb

```diff
@@ -24,10 +24,13 @@ def initialize(action_type)
 
       def build_json_post_arguments
         case @action_type
-        when ActionType::CREATE, ActionType::APPLY, ActionType::DELETE
-          if !@request_arguments.key?(:cloud_providers)
-            @request_arguments[:cloud_providers] = []
-          end
+        when ActionType::CREATE
+          @request_arguments[:name] = random_stack_name if !@request_arguments.key?(:name)
+          @request_arguments[:cloud_providers] = [] if !@request_arguments.key?(:cloud_providers)
+          @request_arguments[:parameters] = [] if !@request_arguments.key?(:parameters)
```

Comment:
Same with these you can do e.g.
```suggestion
          @request_arguments[:cloud_providers] ||= []
          @request_arguments[:parameters]      ||= []
```